auth/auth.go

```diff
@@ -114,7 +120,15 @@ func NewClient(ctx context.Context, conf *internal.AuthConfig) (*Client, error)
 		return nil, err
 	}
 
-	transport, _, err := transport.NewHTTPClient(ctx, conf.Opts...)
+	var opts []option.ClientOption
```

Comment:
You must have tried this already... but just for my curiosity, can we access `conf.Opts` here without creating a new `[]option.ClientOption` and append the `ts`?

Reply:
`conf` gets passed into this function as an argument. I'm not a huge fan of modifying function arguments, since the caller usually has a reference to the same struct.